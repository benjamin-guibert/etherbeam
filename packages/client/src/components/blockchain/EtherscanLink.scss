@use '../../styles/resources/sizes';

.my-etherscanlink-img {
  display: inline-block;
  background-color: #ffffff;
  border-radius: sizes.$border-radius;
  vertical-align: middle;
  padding: 1px;
  width: sizes.$size;
  height: sizes.$size;

  img {
    width: 100%;
    height: 100%;
    margin: auto;
  }
}
