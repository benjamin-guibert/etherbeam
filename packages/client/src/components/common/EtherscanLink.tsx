import React, { ReactElement } from 'react'
import Button from 'react-bootstrap/Button'
import Icon from './Icon'
import EtherscanIcon from 'images/etherscan.svg'
import EtherscanWhiteIcon from 'images/etherscan-white.svg'

interface EtherscanLinkProps {
  url: string
  white?: boolean
  size?: 'xs' | 'sm' | 'm' | 'lg' | 'xl' | number
  className?: string
  noVerticalAlignment?: boolean
}

const EtherscanLink = ({
  url,
  white,
  size = 'lg',
  className,
  noVerticalAlignment,
}: EtherscanLinkProps): ReactElement => {
  return (
    <Button
      className={['shadow-none', className].join(' ')}
      variant="link"
      title="Link to Etherscan"
      href={url}
      onClick={(event) => event.stopPropagation()}
      target="_blank"
    >
      <Icon
        src={white ? EtherscanWhiteIcon : EtherscanIcon}
        size={size}
        alt="Link to Etherscan"
        noVerticalAlignment={noVerticalAlignment}
      />
    </Button>
  )
}

export default EtherscanLink
