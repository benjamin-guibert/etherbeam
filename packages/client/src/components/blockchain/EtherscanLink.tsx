import React, { ReactElement } from 'react'
import Link from 'components/Link'
import './EtherscanLink.scss'

interface EtherscanLinkProps {
  url: string
}

const EtherscanLink = ({ url }: EtherscanLinkProps): ReactElement => {
  return (
    <Link href={url} title="Etherscan" blank>
      <div className="my-etherscanlink-img">
        <img src="/images/etherscan.svg" />
      </div>
    </Link>
  )
}

export default EtherscanLink
