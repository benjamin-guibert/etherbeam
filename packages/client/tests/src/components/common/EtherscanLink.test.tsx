import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'
import EtherscanLink from 'components/common/EtherscanLink'

describe('<EtherscanLink />', () => {
  let component: ReactWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = mount(
        <EtherscanLink
          url="https://etherscan.io/address/0x0000000000000000000000000000000000000111"
          size="sm"
          className="app-class"
        />
      )
    })

    it('should render button', () => {
      const button = component.find('Button')
      expect(button.prop('href')).toBe('https://etherscan.io/address/0x0000000000000000000000000000000000000111')
      expect(button.hasClass('app-class')).toBeTruthy()
    })

    it('should render icon', () => {
      const icon = component.find('Icon')
      expect(icon.prop('size')).toBe('sm')
      expect(icon.prop('alt')).toBe('Link to Etherscan')
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Without vertical alignment', () => {
    beforeAll(() => {
      component = mount(
        <EtherscanLink
          url="https://etherscan.io/address/0x0000000000000000000000000000000000000111"
          size="sm"
          className="app-class"
          noVerticalAlignment
        />
      )
    })

    it('should render button', () => {
      const button = component.find('Button')
      expect(button.prop('href')).toBe('https://etherscan.io/address/0x0000000000000000000000000000000000000111')
      expect(button.hasClass('app-class')).toBeTruthy()
    })

    it('should render icon', () => {
      const icon = component.find('Icon')
      expect(icon.prop('size')).toBe('sm')
      expect(icon.prop('alt')).toBe('Link to Etherscan')
      expect(icon.prop('noVerticalAlignment')).toBeTruthy()
    })
  })
})
