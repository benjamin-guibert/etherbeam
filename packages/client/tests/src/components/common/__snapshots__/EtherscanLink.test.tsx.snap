// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EtherscanLink /> Default should match snapshot 1`] = `
<a
  class="shadow-none app-class btn btn-link"
  href="https://etherscan.io/address/0x0000000000000000000000000000000000000111"
  target="_blank"
  title="Link to Etherscan"
>
  <img
    alt="Link to Etherscan"
    class="app-icon app-icon-sm"
    src=""
  />
</a>
`;
